test/unit/base-api-client.spec.ts

```diff
@@ -0,0 +1,147 @@
+import { ApiRequestHandler } from '../../src/api-request-handler';
+import { BaseApiClient } from '../../src/base-api-client';
+
+jest.mock('../../src/api-request-handler');
+
+interface TJob {
+    id: string;
+};
+interface TResult { };
+
+describe('base-api-client', () => {
+    let sut: BaseApiClient<TJob, TResult>;
+    let mockMakeApiRequest: jest.Mock;
+
+    const jobId = 'Umx5c6F7pH7r';
+    const otherJobId = 'EMx5c67p3dr';
+    const jobDetails = {
+        id: jobId,
+        status: 'completed',
+        created_on: '2018-05-05T23:23:22.29Z'
+    } as TJob;
+
+    beforeEach(() => {
+        mockMakeApiRequest = jest.fn();
+        (ApiRequestHandler as jest.Mock<ApiRequestHandler>).mockImplementationOnce(() => ({
+            makeApiRequest: mockMakeApiRequest
+        }));
+        sut = new BaseApiClient('testtoken', 'topic_extraction');
+    });
+
+    describe('_getJobDetails', () => {
+        it('get job by id', async () => {
+            mockMakeApiRequest.mockResolvedValue(jobDetails);
+
+            const job = await (sut as any)._getJobDetails(jobId);
```

Comment:
Since these are protected methods, we do this cast to `any`. Open to feedback on this

Reply:
We've discussed this and it seems reasonable to me. I'd like @dmtrrk or @k-weng to look

Reply:
briefly:
- base client should be abstract
- we don't need tests for abstract class, but we should test only public contacts

Reply:
Aren't all of these methods effectively tested by `topic-extraction-client.spec.ts`?